File: src/main/java/tech/zolhungaj/amqapi/servercommands/social/DirectMessage.java
package tech.zolhungaj.amqapi.servercommands.social;

import tech.zolhungaj.amqapi.servercommands.CommandType;
import tech.zolhungaj.amqapi.servercommands.objects.messages.DirectMessageEmojis;


/**
 * Event that happens when receiving a direct message
 */
@CommandType("chat message")
public record DirectMessage(
        DirectMessageEmojis emojis,
        String sender,
        String message,
        Boolean modMessage
){}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/social/DirectMessage.kt
package tech.zolhungaj.amqapi.servercommands.social

import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.messages.DirectMessageEmojis

@JvmRecord
@CommandType("chat message")
data class DirectMessage(
    val emojis: DirectMessageEmojis,
    val sender: String,
    val message: String,
    val modMessage: Boolean
)
